modules/pricer/src/main/java/com/opengamma/strata/pricer/rate/bond/DiscountingFixedCouponBondTradePricer.java

```diff
@@ -0,0 +1,1076 @@
+   *  if false, the spread is added to continuously compounded rates
+   * @param periodsPerYear  the number of periods per year
+   * @return the present value of the fixed coupon bond trade
+   */
+  public CurrencyAmount presentValueProductWithZSpread(
+      FixedCouponBondTrade trade,
+      LegalEntityDiscountingProvider provider,
+      double zSpread,
+      boolean periodic,
+      int periodsPerYear) {
+    FixedCouponBond product = trade.getProduct();
+    ExpandedFixedCouponBond expanded = product.expand();
+    LocalDate settlementDate = trade.getTradeInfo().getSettlementDate().get();
+    IssuerCurveDiscountFactors discountFactors = provider.issuerCurveDiscountFactors(
+        product.getLegalEntityId(), product.getCurrency());
+    CurrencyAmount pvNominal = presentValueNominalFromZSpread(
+        expanded, discountFactors, zSpread, periodic, periodsPerYear, settlementDate);
+    CurrencyAmount pvCoupon = presentValueCouponFromZSpread(expanded, discountFactors, zSpread, periodic,
+        periodsPerYear, settlementDate, product.getExCouponPeriod().getDays() == 0);
+    return pvNominal.plus(pvCoupon);
+  }
+
+  /**
+   * Calculates the present value of the fixed coupon bond trade from its clean price.
+   * <p>
+   * The present value of the trade is the value on the valuation date.
+   * The result is expressed using the payment currency of the bond.
+   * <p>
+   * The computation uses {@code settlementDate}. Thus {@code tradeInfo} in the trade must not be empty.
+   * 
+   * @param trade  the trade to price
+   * @param provider  the rates provider
+   * @param cleanPrice  the clean price.
+   * @return the present value of the fixed coupon bond trade
+   */
+  public CurrencyAmount presentValueFromCleanPrice(
+      FixedCouponBondTrade trade,
+      LegalEntityDiscountingProvider provider,
+      double cleanPrice) {
+
+    CurrencyAmount pvProduct = presentValueProductFromCleanPrice(trade, provider, cleanPrice);
+    CurrencyAmount pvPayment = presentValuePayment(trade, provider);
+    return pvProduct.plus(pvPayment);
+  }
+
+  /**
+   * Calculates the present value of the fixed coupon bond product under the specified trade from its clean price.
+   * <p>
+   * The present value of the trade is the value on the valuation date.
+   * The result is expressed using the payment currency of the bond.
+   * <p>
+   * The computation uses {@code settlementDate}. Thus {@code tradeInfo} in the trade must not be empty.
+   * 
+   * @param trade  the trade to price
+   * @param provider  the rates provider
+   * @param cleanPrice  the clean price.
+   * @return the present value of the fixed coupon bond trade
+   */
+  public CurrencyAmount presentValueProductFromCleanPrice(
+      FixedCouponBondTrade trade,
+      LegalEntityDiscountingProvider provider,
+      double cleanPrice) {
+    FixedCouponBond product = trade.getProduct();
+    LocalDate settlementDate = trade.getTradeInfo().getSettlementDate().get();
```

Comment:
The pvFromCleanPrice should apply at the standard settlement date of the bond security, not at trade settlement date.